PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -0,0 +1,687 @@
+        fillHistos<0>(lambda_1);
+        partSingle<2, 0>(lambda_1);
+
+        // lambda-antilambda
+        for (auto const& antilambda_1 : anti_lambda_tracks) {
+          if (antilambda_1.mass() > lambda_left[0] && antilambda_1.mass() < lambda_left[1]) {
+            partPair<2, 0>(lambda_1, antilambda_1);
+          }
+        }
+
+        // lambda-lambda
+        for (auto const& lambda_2 : lambda_tracks) {
+          if ((lambda_2.index() != lambda_1.index()) && (lambda_1.postrackid() != lambda_2.postrackid()) && (lambda_1.negtrackid() != lambda_2.negtrackid())) {
+            if (lambda_2.mass() > lambda_left[0] && lambda_2.mass() < lambda_left[1]) {
+              partPair<2, 1>(lambda_1, lambda_2);
+            }
+          }
+        }
+      }
+    }
+
+    // Anti-Lambda
+    for (auto const& antilambda_1 : anti_lambda_tracks) {
+
+      // Anti-Lambda Mass Window
+      if (antilambda_1.mass() > lambda_mass[0] && antilambda_1.mass() < lambda_mass[1]) {
+        fillHistos<1>(antilambda_1);
+        partSingle<0, 1>(antilambda_1);
+        ++nLaM;
+        ++nLa;
+
+        // antilambda - antilambda
+        for (auto const& antilambda_2 : anti_lambda_tracks) {
+          if ((antilambda_2.index() != antilambda_1.index()) && (antilambda_1.postrackid() != antilambda_2.postrackid()) && (antilambda_1.negtrackid() != antilambda_2.negtrackid())) {
+            if (antilambda_2.mass() > lambda_mass[0] && antilambda_2.mass() < lambda_mass[1]) {
+              partPair<0, 2>(antilambda_1, antilambda_2);
+            }
+          }
+        }
+      }
+
+      // Anti-Lambda Right Sideband
+      if (antilambda_1.mass() > lambda_right[0] && antilambda_1.mass() < lambda_right[1]) {
+        fillHistos<1>(antilambda_1);
+        partSingle<1, 1>(antilambda_1);
+
+        // antilambda - antilambda
+        for (auto const& antilambda_2 : anti_lambda_tracks) {
+          if ((antilambda_2.index() != antilambda_1.index()) && (antilambda_1.postrackid() != antilambda_2.postrackid()) && (antilambda_1.negtrackid() != antilambda_2.negtrackid())) {
+            if (antilambda_2.mass() > lambda_right[0] && antilambda_2.mass() < lambda_right[1]) {
+              partPair<1, 2>(antilambda_1, antilambda_2);
+            }
+          }
+        }
+      }
+
+      // Anti-Lambda Left Sideband
+      if (antilambda_1.mass() > lambda_left[0] && antilambda_1.mass() < lambda_left[1]) {
+        fillHistos<1>(antilambda_1);
+        partSingle<2, 1>(antilambda_1);
+
+        // antilambda - antilambda
+        for (auto const& antilambda_2 : anti_lambda_tracks) {
+          if ((antilambda_2.index() != antilambda_1.index()) && (antilambda_1.postrackid() != antilambda_2.postrackid()) && (antilambda_1.negtrackid() != antilambda_2.negtrackid())) {
+            if (antilambda_2.mass() > lambda_left[0] && antilambda_2.mass() < lambda_left[1]) {
+              partPair<2, 2>(antilambda_1, antilambda_2);
+            }
+          }
+        }
+      }
+    }
```

Comment:
In here two of my previous comments apply

Using 0, 1, 2, ... specially when the same number has different meaning is confusing, error prone, difficult to maintain, and difficult to evolve
I would suggest to define an `enum` for each context, for instance
```
enum MassWindowType {
  kCentralWindow = 0,
  kLeftWindow,
  kRightWindow,
  kNoOfMassWindows
}
```
the latest one in case you consider the possibility to iterate across mass windows (and it is always a good practice), for the mass window context and the likes for single particles and pair combinations.

Then using those `enum`s as the corresponding parameter types of the templated functions will immediately clarify their use and avoid ambiguities and not easy to catch mistakes

Basically the same code is replicated several times and as said replicating code is error prone and difficult to maintain/evolve. I would suggest trying to parameterize what is needed to be replicated in order to not replicate it